### PR TITLE
OMG-218 refactor: vm stats

### PR DESCRIPTION
**Pull request by InoMurko**
This is now the complete VM overview. 
I've added :vm_stats dependency which I've vetted myself. It's written and maintained by Fred Hebert, @mononcqc. It has no other dependencies - because it's just calling VM modules and does the necessary grouping. It works by simply enforcing a behavior  which we need to implement and send off to AppSignal.

I've also rewritten the plotting part. It now makes more sense :D

```
title: VM dashboard
graphs:
  - title: 'Writes, reads and multireads - LevelDB in interval'
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: leveldb_write
        fields:
          - COUNTER
        tags:
          node: '*'
      - name: leveldb_read
        fields:
          - COUNTER
        tags:
          node: '*'
      - name: leveldb_multiread
        fields:
          - COUNTER
        tags:
          node: '*'
  - title: vmstats$.scheduler_wall_time$.1.total - active
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.scheduler_wall_time$.1$.total
        fields:
          - GAUGE
        tags:
          node: '*'
      - name: vmstats$.scheduler_wall_time$.1$.active
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.scheduler_wall_time$.2.total - active
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.scheduler_wall_time$.2$.total
        fields:
          - GAUGE
        tags:
          node: '*'
      - name: vmstats$.scheduler_wall_time$.2$.active
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.scheduler_wall_time$.3.total - active
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.scheduler_wall_time$.3$.total
        fields:
          - GAUGE
        tags:
          node: '*'
      - name: vmstats$.scheduler_wall_time$.3$.active
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.scheduler_wall_time$.4.total - active
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.scheduler_wall_time$.4$.total
        fields:
          - GAUGE
        tags:
          node: '*'
      - name: vmstats$.scheduler_wall_time$.4$.active
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.scheduler_wall_time$.5.total - active
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.scheduler_wall_time$.5$.total
        fields:
          - GAUGE
        tags:
          node: '*'
      - name: vmstats$.scheduler_wall_time$.5$.active
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.scheduler_wall_time$.6.total - active
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.scheduler_wall_time$.6$.total
        fields:
          - GAUGE
        tags:
          node: '*'
      - name: vmstats$.scheduler_wall_time$.6$.active
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.scheduler_wall_time$.7.total - active
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.scheduler_wall_time$.7$.total
        fields:
          - GAUGE
        tags:
          node: '*'
      - name: vmstats$.scheduler_wall_time$.7$.active
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.scheduler_wall_time$.8.total - active
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.scheduler_wall_time$.8$.total
        fields:
          - GAUGE
        tags:
          node: '*'
      - name: vmstats$.scheduler_wall_time$.8$.active
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.reductions
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.reductions
        fields:
          - COUNTER
        tags:
          node: '*'
  - title: vmstats$.io$.bytes_out
    line_label: '%node%'
    format: size
    format_input: byte
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.io$.bytes_out
        fields:
          - COUNTER
        tags:
          node: '*'
  - title: vmstats$.io$.bytes_in
    line_label: '%node%'
    format: size
    format_input: byte
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.io$.bytes_in
        fields:
          - COUNTER
        tags:
          node: '*'
  - title: vmstats$.gc$.words_reclaimed
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.gc$.words_reclaimed
        fields:
          - COUNTER
        tags:
          node: '*'
  - title: vmstats$.gc$.count
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.gc$.count
        fields:
          - COUNTER
        tags:
          node: '*'
  - title: vmstats$.run_queue
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.run_queue
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.modules
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.modules
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.port_count
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.port_count
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.port_limit
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.port_limit
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.proc_count
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.proc_count
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.proc_limit
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.proc_limit
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.messages_in_queues
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.messages_in_queues
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.memory$.total
    line_label: '%node%'
    format: size
    format_input: byte
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.memory$.total
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.memory$.procs_used
    line_label: '%node%'
    format: size
    format_input: byte
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.memory$.procs_used
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.memory$.ets
    line_label: '%node%'
    format: size
    format_input: byte
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.memory$.ets
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.memory$.binary
    line_label: '%node%'
    format: size
    format_input: byte
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.memory$.binary
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.memory$.atom_used
    line_label: '%node%'
    format: size
    format_input: byte
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.memory$.atom_used
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.error_logger_queue_len
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.error_logger_queue_len
        fields:
          - GAUGE
        tags:
          node: '*'
  - title: vmstats$.atom_count
    line_label: '%node%'
    format: number
    draw_null_as_zero: true
    metrics:
      - name: vmstats$.atom_count
        fields:
          - GAUGE
        tags:
          node: '*'
```

### CODE REVIEW

**Comment by jbunce** [comment]
Going to let somebody who knows Elixir better than me to hit the approve button :)

File: apps/omg_status/lib/status.ex [hunk 1]
```diff
@@ -23,53 +23,23 @@ defmodule OMG.Status do
...
-    Enum.map(vm_metrics(), fn {name, invoke} ->
-      Recorder.prepare_child(%Recorder{
-        name: name,
-        fn: invoke,
-        reporter: &Appsignal.set_gauge/3
-      })
-    end)
-    |> Supervisor.start_link(strategy: :one_for_one, name: Status.Supervisor)
+    children =
+      case is_enabled?() do
```
Comment (pnowosie) [inline apps/omg_status/lib/status.ex]: case with booleans looks strange to me :)
how about
```
Application.put_env(:vmstats, :sink, Status.Metric.Recorder)
children = if is_enabled?(), do: [Recorder.prepare_child()], else: []

File: config/config.exs [hunk 2]
```diff
@@ -23,5 +23,3 @@ config :appsignal, :config,
   name: "OmiseGO Plasma MoreVP Implementation",
   env: Mix.env(),
   active: true
-
-import_config "#{Mix.env()}.exs"
```
Comment (pnowosie) [inline config/config.exs]: This looks suspicious, why we don't want to include specific env's config

File: apps/omg_status/lib/status.ex (hunk 1 above)
Comment (InoMurko) [reply apps/omg_status/lib/status.ex]: Haha, yeah ... I come from erlang where everything is pattern matching

**Pull request merged**